### ISSUE TITLE
Remove ruleset specific enumerations from the overall schema

### ISSUE THREAD

**Issue by nealkruis** [open229/ruleset-model-description-schema#180]
We can figure out ways to "forward declare" enumerations (so that the specific ruleset defines them), or maybe generate a ruleset specific schema from a template where some enumerations and data groups are inserted when the schema is "compiled".

These enumerations are not within the scope of 229, and often not within the scope of ASHRAE to decide.

**Comment by nealkruis** [comment]
Yes. Through the [definitions](https://bigladder.github.io/ruleset-data-schema/specifications/buildingperformanceruleset/#definitions) of a Machine Readable Ruleset.

**Comment by JasonGlazer** [comment]
I'm not sure I'm following the example provided. What makes this a "forward declare" type of enumeration?

**Comment by nealkruis** [comment]
By "forward declare", I mean that the primary schema would reference an enumeration (e.g., <SpaceType>), without defining it, because the actual definition would come from an external machine readable ruleset. You'd have a script that does something like:

`generate_229_schema(primary_schema_template, machine_readable_ruleset)` that would pull the ruleset specific enumeration definitions into a complete schema specific to the ruleset.

**Comment by nealkruis** [comment]
For now, I think you can create two separate files: A primary schema "template" with the forward declarations, and a second file with ruleset specific enumerations / data groups. You can reference the latter from the former for now and develop the script to generate them more dynamically later.

**Comment by JasonGlazer** [comment]
If you could create some example text, that would help me. It sounds like you are describing essentially what we have now except for something that would indicate the forward declaration itself.
